tests/integration/url_generator_only/test_url_generator_only.py

```diff
@@ -156,38 +168,137 @@ def test_function_imports():
+    random.seed(10)
+    methodLists = ["POST", "MERGE", "PUT", "DELETE", "GET"]
+    for method in methodLists:
+        entities , queryable_factory = builder_with_restrictions(method)
+        option_list_filter_query = []
+        option_list_filter_query.clear()
+        for queryable in entities:
+            entityset_urls_count = len(queryable.entity_set.entity_type.proprties())
+            for _ in range(entityset_urls_count):
+                q = queryable_factory(queryable, logger, 1)
+                queries,body = q.generate()
+                for option_query in queries.options_strings:
+                    if option_query == "$filter":
+                        if len(queries.options_strings[option_query]) != 0:
+                              option_list_filter_query.append(queries.options_strings[option_query])
+                        
+        if method == "POST" or method == "PUT" or method == "MERGE" or method == "DELETE":
+            assert option_list_filter_query == []
+        else:
+            for filter_queries in option_list_filter_query:
+                assert "ProductID" not in filter_queries
+
+def test_direct_builder_orderby_query_option():
+    random.seed(10)
+    methodLists = ["POST", "MERGE", "PUT", "DELETE", "GET"]
+    for method in methodLists:
+        entities , queryable_factory = builder_with_restrictions(method)
+        option_list_orderby_query = []
+        option_list_orderby_query.clear()
+        for queryable in entities:
+            entityset_urls_count = len(queryable.entity_set.entity_type.proprties())
+            for _ in range(entityset_urls_count):
+                q = queryable_factory(queryable, logger, 1)
+                queries,body = q.generate()
+                for option_query in queries.options_strings:
+                    if option_query == "$orderby":
+                        if len(queries.options_strings[option_query]) != 0:
+                              option_list_orderby_query.append(queries.options_strings[option_query])
+          
+        if method == "POST" or method == "PUT" or method == "MERGE" or method == "DELETE":
+            assert option_list_orderby_query == []
+        else:
+            assert "ProductID" not in option_list_orderby_query
+
+@pytest.mark.parametrize('method_name,URI', [
+    ("POST", "Suppliers?sap-client=500"),
+    ("MERGE", "Suppliers(SupplierID=1564408389)?sap-client=500"),
+    ("PUT", "Order_Details_Extendeds(OrderID=-717893126,ProductID=349786362,ProductName='8%C2%A3T%E2%80%A0%C2%81%C3%A1C%C3%8F%C3%96%C3%A5%C5%92%C2%A7%29%C3%83T%C3%9A%C5%92m',UnitPrice=16647857647390.73m,Quantity=-1289,Discount=1.4474711818031789e+20f)?sap-client=500"),
+    ("DELETE", "Alphabetical_list_of_products?sap-client=500")
+])
+
+def test_direct_builder_Uri_unrestricted(method_name, URI):
+    random.seed(30)
+    entities , queryable_factory = builder(method_name)
+    methodList = []
+    methodList.clear()
```

Comment:
is this needed? every parametrized test should be new instance with new clear methodList from line 226

Reply:
This won't be required, will remove this in the next commit..